scripts/update_docs_toc.py

```diff
@@ -0,0 +1,79 @@
+# /// script
+# requires-python = ">=3.13"
+# dependencies = ["markdown~=3.8", "mkdocs~=1.6.1"]
```

Comment:
instead of pinning dependencies here, maybe we could use a script lockfile, similar to [what we do for mdtest.py in the Ruff repo](https://github.com/astral-sh/ruff/blob/main/crates/ty_python_semantic/mdtest.py.lock) (more dogfooding!)